scrapy/commands/crawl.py

```diff
@@ -54,8 +54,16 @@ def run(self, args, opts):
             raise UsageError("running 'scrapy crawl' with more than one spider is no longer supported")
         spname = args[0]
 
-        self.crawler_process.crawl(spname, **opts.spargs)
+        res = self.crawler_process.crawl(spname, **opts.spargs)
         self.crawler_process.start()
 
-        if self.crawler_process.bootstrap_failed:
+        if hasattr(res, 'result') and res.result is not None and issubclass(res.result.type, Exception):
             self.exitcode = 1
+        else:
+            self.crawler_process.start()
```

Comment:
So this will run twice?

Reply:
sorry, I didn't notice it. I've removed the redundant code. it should start if no exception after executing self.crawler_process.crawl()